### PR TITLE
Improve readability of config parsing and use ENV vars in example

### PR DESCRIPTION
**Pull request by cbeneke**
Signed-off-by: Christian Beneke <c.beneke@wirelab.org>

### CODE REVIEW

File: pkg/fipcontroller/controller.go [hunk 1]
```diff
@@ -41,77 +42,87 @@ func NewController(config *Configuration) (*Controller, error) {
...
 }
 
 func NewControllerConfiguration() (*Configuration, error) {
 	var config Configuration
 
+	// Read config from file if present
 	file, err := ioutil.ReadFile("config/config.json")
-	if err != nil {
-		return nil, fmt.Errorf("could not open config file: %v", err)
+	if err == nil {
```
Comment (metalmatze) [inline pkg/fipcontroller/controller.go]: This should be `if err != nil {`. That way you can early return if you're unable to read the file or unable to unmarshal.

```go

	file, err := ioutil.ReadFile("config/config.json")
	if err != nil {
		return nil, fmt.Errorf("failed to read config: %v", err)
	}
	if err := json.Unmarshal(file, &config); err != nil {
		return nil, fmt.Errorf("could not decode config: %v", err)
	}
```

File: pkg/fipcontroller/controller.go [hunk 2]
```diff
@@ -41,77 +42,87 @@ func NewController(config *Configuration) (*Controller, error) {
...
 	}
 
-	err = json.Unmarshal(file, &config)
-	if err != nil {
-		return nil, fmt.Errorf("could not decode config: %v", err)
+	// overwrite existing configs from ENV vars if present
+	if apiToken := os.Getenv("HETZNER_CLOUD_API_TOKEN"); apiToken != "" {
+		config.HetznerAPIToken = apiToken
+	}
+	if floatingIP := os.Getenv("HETZNER_CLOUD_FLOATING_IP"); floatingIP != "" {
```
Comment (metalmatze) [inline pkg/fipcontroller/controller.go]: This should be a flag.

File: pkg/fipcontroller/controller.go [hunk 3]
```diff
@@ -41,77 +42,87 @@ func NewController(config *Configuration) (*Controller, error) {
...
+		config.FloatingIPAddress = floatingIP
+	}
+	if nodeName := os.Getenv("KUBERNETES_NODE_NAME"); nodeName != "" {
```
Comment (metalmatze) [inline pkg/fipcontroller/controller.go]: This should be a flag.

File: pkg/fipcontroller/controller.go [hunk 4]
```diff
@@ -41,77 +42,87 @@ func NewController(config *Configuration) (*Controller, error) {
...
+		config.NodeName = nodeName
+	}
+	if nodeAddressType := os.Getenv("KUBERNETES_NODE_ADDRESS_TYPE"); nodeAddressType != "" {
```
Comment (metalmatze) [inline pkg/fipcontroller/controller.go]: This should be a flag.

File: pkg/fipcontroller/controller.go (hunk 1 above)
Comment (cbeneke) [reply pkg/fipcontroller/controller.go]: The basic idea was that the config file is optional. Rewrote a little :)

File: pkg/fipcontroller/controller.go (hunk 2 above)
Comment (cbeneke) [reply pkg/fipcontroller/controller.go]: I think this should be resolved with https://github.com/cbeneke/hcloud-fip-controller/pull/13/commits/c291e62932140c4237a3ebad0eb7149c9c8586db

File: pkg/fipcontroller/controller.go (hunk 4 above)
Comment (cbeneke) [reply pkg/fipcontroller/controller.go]: Should be resolved with https://github.com/cbeneke/hcloud-fip-controller/pull/13/commits/c291e62932140c4237a3ebad0eb7149c9c8586db

File: pkg/fipcontroller/controller.go (hunk 3 above)
Comment (cbeneke) [reply pkg/fipcontroller/controller.go]: Should be resolved with https://github.com/cbeneke/hcloud-fip-controller/pull/13/commits/c291e62932140c4237a3ebad0eb7149c9c8586db

**Comment by cbeneke** [comment]
Thanks for the review, I have updated the code further.. not quite sure if thats what you meant exactly though ;) Could you have another look? Thanks!

**Pull request merged**